src/ray/stats/metric_exporter_client.cc

```diff
@@ -32,5 +38,35 @@ void MetricExporterDecorator::ReportMetrics(const std::vector<MetricPoint> &poin
     exporter_->ReportMetrics(points);
   }
 }
+
+///
+/// Metrics Agent Exporter
+///
+MetricsAgentExporter::MetricsAgentExporter(std::shared_ptr<MetricExporterClient> exporter,
+                                           const int port,
+                                           boost::asio::io_service &io_service)
+    : MetricExporterDecorator(exporter), client_call_manager_(io_service) {
+  const std::string address = "127.0.0.1";
+  client_.reset(new rpc::MetricsAgentClient(address, port, client_call_manager_));
+}
+
+void MetricsAgentExporter::ReportMetrics(const std::vector<MetricPoint> &points) {
```

Comment:
This will be called periodically or when the batch size is reached, right?

Reply:
Actullay it's called by opencensus handler periodically. @edoakes

Reply:
I actually have a question about this part https://github.com/ray-project/ray/blob/fa1eadd61fc72bd28420494ea5e0a6bb53306bf5/src/ray/stats/metric_exporter.h#L72. Doesn't it mean we will lose data if there are more than batch size metrics-points?

Reply:
It is exported periodically (currently once per 10 seconds) by opencensus handler. At each period, it can call reporter more than once if there are more points than the batch size.